### PR TITLE
Major/go v2 sdk (#2)

### PR DESCRIPTION
**Pull request by hectorruiz-it**
# CHANGES

## GENERAL

- Standarized letme configuration files to `.ini` to match aws credentials.
- Upgraded dependencies to Go 1.22
- Upgrade go SDK to v2.
- Version name is now New Horizons.

## OBTAIN

**Refactor**:

- Extracted and simplified logic from main function.
- Created two functions:
  - `AssumeRoleChained`
  - `AssumeRole`
- Stored new credentials through `ini` library for ease of use and validation.
-  Created `LoadAwsCredentials` and `LoadAwsConfig` to store credentials on `.aws/credentials` and `.aws/config` respectively.
- Change DynamoDB query from Scan to Get (**requires name as primary key**).
- Validation of credentials now doesn't require to use assume rol.
- Credentials v1 is supported for credentials without MFA.

## CONFIG && CONFIG-FILE

- Update functions to use `ini` library.
- Create `ini` files instead of `toml`.

## LIST

- Extracted  logic from main function.
- Fixed table length on output. Now is calculated based on the item.Name with max length.
- Removed local flag as it's unused.

## UTILS

- Removed non-used functions.
- Removed non-used structs.

## REMOVE

- Simplified validations.
  - Instead of checking an array now checks values on a map.
  - For verifying that a context is managed by `letme`, now uses the `ini` module to check if the comment matches the one used by `letme`.

### CODE REVIEW

**Comment by hectorruiz-it** [comment]
- [ ] Remove references to `hectorruiz-it` and `letme-alpha` directory.
- [ ] Update README.md

**Comment by lockedinspace** [comment]
Hi @hectorruiz-it , thanks for your contribution. I will be looking at this as soon as I can.

Best,

File: go.mod [hunk 1]
```diff
@@ -1,20 +1,37 @@
-module github.com/lockedinspace/letme
+module github.com/github.com/lockedinspace/letme
```
Comment (lockedinspace) [inline go.mod]: Seems like the ``module github.com/github.com/lockedinspace/letme`` line has duplicated github.com entry.

Comment (hectorruiz-it) [reply go.mod]: Fixed on next commit

**Review by lockedinspace — APPROVED** [review]

**Pull request merged**